### PR TITLE
backup: config adjust for checkpoint

### LINKED ISSUE #39632: checkpoint for backup now is not compatible with some experimental features
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. run br with --lastbackupts or --use-backupmeta-v2

### 2. What did you expect to see? (Required)
the checkpoint mode is off
### 3. What did you see instead (Required)
the checkpoint mode is on
### 4. What is your TiDB version? (Required)
6.5

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #39632

Problem Summary:
Need to off the checkpoint when use some experimental feature.
### What is changed and how it works?

close off the checkpoint mode when one of these feature is used
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by joccau — APPROVED** [review]
LGTM

File: br/pkg/task/backup.go [hunk 1]
```diff
@@ -159,11 +159,22 @@ func (cfg *BackupConfig) ParseFromFlags(flags *pflag.FlagSet) error {
...
+	if err != nil {
+		return errors.Trace(err)
+	}
+	cfg.UseCheckpoint, err = flags.GetBool(flagUseCheckpoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	cfg.UseCheckpoint = !noCheckpoint
+	if cfg.LastBackupTS > 0 {
+		// TODO: compatible with incremental backup
```
Comment (3pointer) [inline br/pkg/task/backup.go:171]: plz add some log here.

File: br/pkg/task/backup.go [hunk 2]
```diff
@@ -159,11 +159,22 @@ func (cfg *BackupConfig) ParseFromFlags(flags *pflag.FlagSet) error {
...
+		cfg.UseCheckpoint = false
+	}
+	if cfg.UseBackupMetaV2 {
+		// TODO: compatible with backup meta v2, maybe just clean the meta files
```
Comment (3pointer) [inline br/pkg/task/backup.go:175]: ditto

**Review by 3pointer — COMMENTED** [review]
rest LGTM

**Review by 3pointer — APPROVED** [review]

**Pull request merged**